install/shared_disk_check.sh

```diff
@@ -0,0 +1,53 @@
+#! /bin/bash
+
+PUBLIC_NFS_IPS=()
+
+json=$(<config.json)
+
+nodes=("HEADNODE_IP" "CLIENTNODE_IP")
+for node in "${nodes[@]}"; do
+    ips=($(echo $json | jq -r ".${node}[]"))
+    for ip in "${ips[@]}"; do
+        PUBLIC_NFS_IPS+=("$ip")
+    done
+done
+
+# NFS Server Configuration
+PUBLIC_NFS_SERVER_IP="${PUBLIC_NFS_IPS[0]}"
+
+# Array of NFS Client IPs
+PUBLIC_NFS_CLIENT_IPS=("${PUBLIC_NFS_IPS[@]:1}")
+
+
+# User
+USERNAME=$admin_name
+shared_file="$(jq -r '.shared_dir' config.json)/nodes_status"
+
+# Headnode connectivity check
+ssh -o StrictHostKeyChecking=no "$USERNAME"@$PUBLIC_NFS_SERVER_IP <<EOF
+    echo "$PUBLIC_NFS_SERVER_IP | success" | sudo tee $shared_file
+EOF
+
+# Client node connectivity check
+for CLIENT_IP in "${PUBLIC_NFS_CLIENT_IPS[@]}"; do
+    ssh -o StrictHostKeyChecking=no "$USERNAME"@$CLIENT_IP <<EOF
+echo "$CLIENT_IP | success" | sudo tee -a $shared_file
+EOF
+done
+
+# copying node_status file into local directory
+local_file="node_status"
```

Comment:
does local_file exist on the user's machine at this point in the code? I don't see where it's created

Reply:
no, the file `node_status` in the server would be fetched and renamed to $local_file in the local directory

Reply:
I would change to be simple